packages/backend-common/src/discovery/HostDiscovery.ts

```diff
@@ -28,13 +28,16 @@ import { readHttpServerOptions } from '@backstage/backend-app-api';
  *
  * @public
  */
-export class SingleHostDiscovery implements PluginEndpointDiscovery {
+
+export class HostDiscovery implements PluginEndpointDiscovery {
   /**
-   * Creates a new SingleHostDiscovery discovery instance by reading
+   * Creates a new HostDiscovery discovery instance by reading
    * from the `backend` config section, specifically the `.baseUrl` for
    * discovering the external URL, and the `.listen` and `.https` config
    * for the internal one.
    *
+   * Can be overridden by in config by providing a `backend.discovery` section with a `<pluginId>` key.
```

Comment:
Noting that I don't think we want this under the backend config key, but rather a top-level `discovery` key, given that we'll want to be consuming this from the frontend as well

Reply:
Addressed in 54ea03b